contracts/Hub.sol

```diff
@@ -0,0 +1,138 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.7;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+/**
+ * @title Hub Contract that coordinates Storage, Fund and Minter contract.
+ * @dev The contract can mint new tokens, check the collateral ratio of addresses
+ * and liquidate addresses if the collateral ratio falls below 150%
+ * @author The Everest team.
+ */
+
+contract Hub is Ownable {
+    // Contract addresses
+    Minter private minterContract;
+    Fund private fundContract;
+    Storage private storageContract;
+
+    address[] private authorizedAddresses;
+    mapping(string => address) private tickersymbolToAddress;
+
+
+    modifier onlyAuthorizedAddresses() {
+        bool isAuthorized = false;
+        for(uint i = 0; i < authorizedAddresses.length; i++) {
+            if(authorizedAddresses[i] == msg.sender) {
+                isAuthorized = true;
+            }
+        }
+        require(isAuthorized);
+        _;
+    }
+
+    constructor() {
+    }
+
+    /**
+     * @notice Add an address to the list of authorized addresses
+     * @param addr address that gets added 
+     */
+    function addAuthorizedAddress(address addr) public onlyOwner{
+        authorizedAddresses.push(addr);
+    }
+
+    /**
+     * @notice updates the address asssociated with an synthAsset
+     * @param tickerSymbol identifier of synthAsset
+     * @param addr new contract address of synthAsset 
+     */
+    function updateContractProxy(string memory tickerSymbol, address addr) public onlyAuthorizedAddresses{
+        tickersymbolToAddress[tickerSymbol] = addr; 
```

Comment:
what is the purpose of this function? Storing a new contract for a synthetic asset? 
what would be the use case of this? when do we need to create a new synth contract for an existing contract (I mean if we have already sTSLA, why do we need to redeploy sTLSA?)
Or is the function about adding new synthetic assets? if so, I suggest a name that makes clear to everyone what's the purpose e.g `registerNewSyntheticAsset`